pkg/controllers/provisioner/suite_test.go

```diff
@@ -0,0 +1,120 @@
+	"github.com/aws/karpenter-core/pkg/operator/controller"
+	"github.com/aws/karpenter-core/pkg/operator/scheme"
+	"github.com/aws/karpenter-core/pkg/test"
+	. "github.com/aws/karpenter-core/pkg/test/expectations"
+)
+
+var provisionerController controller.Controller
+var provisioner *v1alpha5.Provisioner
+var ctx context.Context
+var env *test.Environment
+
+func TestAPIs(t *testing.T) {
+	ctx = TestContextWithLogger(t)
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Controllers/Provisioner")
+}
+
+var _ = BeforeSuite(func() {
+	env = test.NewEnvironment(scheme.Scheme, test.WithCRDs(apis.CRDs...))
+	provisionerController = provcontroller.NewController(env.Client)
+	provisioner = &v1alpha5.Provisioner{
+		ObjectMeta: metav1.ObjectMeta{Name: strings.ToLower(randomdata.SillyName())},
+		Spec: v1alpha5.ProvisionerSpec{
+			Taints: []v1.Taint{
+				{
+					Key:    "key",
+					Effect: v1.TaintEffectNoExecute,
+				},
+			},
+			StartupTaints: []v1.Taint{
+				{
+					Key:    "key",
+					Effect: v1.TaintEffectNoExecute,
+				},
+			},
+			Labels: map[string]string{
+				"keyLabel": "valueLabel",
+			},
+			KubeletConfiguration: &v1alpha5.KubeletConfiguration{
+				MaxPods: ptr.Int32(10),
+			},
+			Annotations: map[string]string{
+				"keyAnnotation": "valueAnnotation",
+			},
+		},
+	}
+})
+
+var _ = AfterSuite(func() {
+	Expect(env.Stop()).To(Succeed(), "Failed to stop environment")
+})
+
+var _ = Describe("Provisioner Static Drift Hash", func() {
+	It("should update the static drift hash when provisioner static felid is updated", func() {
+		ExpectApplied(ctx, env.Client, provisioner)
+		ExpectReconcileSucceeded(ctx, provisionerController, client.ObjectKeyFromObject(provisioner))
+		provisioner = ExpectExists(ctx, env.Client, provisioner)
+
+		expectedHash, err := provisioner.StaticDriftHash()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(provisioner.ObjectMeta.Annotations[v1alpha5.ProvisionerHashAnnotationKey]).To(Equal(expectedHash))
+
+		provisioner.Spec.Labels = map[string]string{"keyLabeltest": "valueLabeltest"}
+		ExpectReconcileSucceeded(ctx, provisionerController, client.ObjectKeyFromObject(provisioner))
+		provisioner = ExpectExists(ctx, env.Client, provisioner)
+
+		expectedHashTwo, err := provisioner.StaticDriftHash()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(provisioner.ObjectMeta.Annotations[v1alpha5.ProvisionerHashAnnotationKey]).To(Equal(expectedHashTwo))
+	})
+	It("should not update the static drift hash when provisioner behavior felid is updated", func() {
+		ExpectApplied(ctx, env.Client, provisioner)
+		ExpectReconcileSucceeded(ctx, provisionerController, client.ObjectKeyFromObject(provisioner))
+		provisioner = ExpectExists(ctx, env.Client, provisioner)
+
+		expectedHash, err := provisioner.StaticDriftHash()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(provisioner.ObjectMeta.Annotations[v1alpha5.ProvisionerHashAnnotationKey]).To(Equal(expectedHash))
+
+		provisioner.Spec.TTLSecondsAfterEmpty = lo.ToPtr(int64(20))
```

Comment:
Similar to the other test, would be nice if we just tested all ignored fields here too.